lib/video_info.rb

```diff
@@ -81,6 +81,13 @@ def self.disabled_provider?(provider)
     disable_providers.map(&:downcase).include?(provider.downcase)
   end
 
+  def self.valid_url?(url)
+    PROVIDERS.select { |p| !disabled_provider?(p) }.any? do |provider|
+      provider_const = Providers.const_get(provider)
+      return true if provider_const.usable?(url)
+    end
+  end
```

Comment:
With the new `enabled_providers` method that would just be:

```suggestion
  def self.valid_url?(url)
    enabled_providers any? { |p| p.usable?(url) }
  end
```